### PR TITLE
Cm 4

### PR DESCRIPTION
**Pull request by J2thatsme**

### CODE REVIEW

**Review by mariusgiger — CHANGES_REQUESTED** [review]
looks good, some comments

File: types/comtac/CM-2/uplink.spec.js [hunk 1]
```diff
@@ -1,69 +1,56 @@
...
 
-      expectEmit(function (type, value) {
+      utils.expectEmits((type, value) => {
         assert.equal(type, "sample");
-
         assert.isNotNull(value);
         assert.typeOf(value.data, "object");
 
-        if (value.topic === "lifecycle") {
+        if (value.topic === "lifecyle") {
```
Comment (mariusgiger) [inline types/comtac/CM-2/uplink.spec.js:46]: correct topic

Comment (mariusgiger) [reply types/comtac/CM-2/uplink.spec.js:46]: is this if still required?

File: types/comtac/CM-4/uplink.spec.js [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+            "011202C604090B40090B40090B40090B40090B40090B40090B40090B40",
+        },
+      };
+
+      utils.expectEmits((type, value) => {
+        assert.equal(type, "sample");
+        assert.isNotNull(value);
+        assert.typeOf(value.data, "object");
+
+        if (value.topic === "lifecyle") {
```
Comment (mariusgiger) [inline types/comtac/CM-4/uplink.spec.js:105]: this if is not required, correct topic

File: types/comtac/CM-4/uplink.spec.js [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+          validate(value.data, lifecycleSchema, { throwError: true });
+        }
+      });
+
+      utils.expectEmits((type, value) => {
+        assert.equal(type, "sample");
+        assert.isNotNull(value);
+        assert.typeOf(value.data, "object");
+
+        if (value.topic === "history") {
```
Comment (mariusgiger) [inline types/comtac/CM-4/uplink.spec.js:129]: this if is not required

Comment (mariusgiger) [reply types/comtac/CM-4/uplink.spec.js:129]: @J2thatsme this is not yet done

File: types/comtac/CM-2/uplink.spec.js [hunk 4]
```diff
@@ -1,96 +1,94 @@
...
+        assert.equal(value.data.humidity, 66.36);
+        assert.equal(value.data.temperaturePT100, 20);
+        assert.equal(value.data.adc1, 500);
+        assert.equal(value.data.adc2, 1096);
+        assert.equal(value.data.lem, 2.506);
+        assert.equal(value.data.brightness, 21);
+        validate(value.data, defaultSchema, { throwError: true });
       });
+      /*
+      utils.expectEmits((type, value) => {
```
Comment (mariusgiger) [inline types/comtac/CM-2/uplink.spec.js:83]: readd this to make the tests work

Comment (J2thatsme) [reply types/comtac/CM-2/uplink.spec.js:83]: ups sorry

**Review by mariusgiger — APPROVED** [review]

**Pull request merged**